mm-go-irckit/userbridge.go

```diff
@@ -138,22 +140,18 @@ func (u *User) handleDirectMessageEvent(event *bridge.DirectMessageEvent) {
 	}
 	text, prefix, suffix, showContext, maxlen := u.handleMessageThreadContext(prefixUser, event.MessageID, event.ParentID, event.Event, event.Text)
 
+	lexer := ""
 	codeBlockBackTick := false
 	codeBlockTilde := false
 	text = wordwrap.String(text, maxlen)
 	lines := strings.Split(text, "\n")
 	for _, text := range lines {
-		if strings.HasPrefix(text, "```") && !codeBlockTilde {
-			codeBlockBackTick = !codeBlockBackTick
-		}
-		if strings.HasPrefix(text, "~~~") && !codeBlockBackTick {
-			codeBlockTilde = !codeBlockTilde
-		}
-		// skip empty lines for anything not part of a code block.
-		if !codeBlockBackTick && !codeBlockTilde && text == "" {
+
+		// TODO: Ideally, we want to read the whole code block and syntax highlight on that, but let's go with per-line for now.
+		text, codeBlockBackTick, codeBlockTilde, lexer = u.formatCodeBlockText(text, prefix, codeBlockBackTick, codeBlockTilde, lexer)
```

Comment:
```suggestion
		text, _, _, _ = u.formatCodeBlockText(text, prefix, codeBlockBackTick, codeBlockTilde, lexer)
```
the other variables aren't used, so don't assign them makes it confusing :)

Reply:
Actually, these are used to preserve if/when code blocks start and end.

Reply:
aargh of course